python/ray/tests/test_failure.py

```diff
@@ -35,13 +34,15 @@ def throw_exception_fct2():
     def throw_exception_fct3(x):
         raise Exception("Test function 3 intentionally failed.")
 
+    p = init_error_pubsub()
```

Comment:
You can put 

```
@pytest.fixture
def error_pubsub():
    p = init_error_pubsub
    yield p
    p.close()

def test_something(ray_start, error_pubsub):
    # and here, you can access p through error_pubsub
```
I think this will gurantee pubsub is always called after ray.init